specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/managedCassandra.json

```diff
@@ -774,657 +693,1717 @@
+                      "description": "List of tokens this node covers.",
+                      "items": {
+                        "type": "string"
+                      }
+                    },
+                    "size": {
+                      "type": "integer",
+                      "format": "int32"
+                    },
+                    "hostID": {
+                      "description": "The network ID of the node.",
+                      "type": "string"
+                    },
+                    "rack": {
+                      "description": "The rack this node is part of.",
+                      "type": "string"
+                    },
+                    "timestamp": {
+                      "description": "The timestamp when these statistics were captured.",
+                      "type": "string"
+                    },
+                    "diskUsedKB": {
+                      "description": "The amount of disk used, in kB, of the directory /var/lib/cassandra.",
+                      "type": "integer",
+                      "format": "int64"
+                    },
+                    "diskFreeKB": {
+                      "description": "The amount of disk free, in kB, of the directory /var/lib/cassandra.",
+                      "type": "integer",
+                      "format": "int64"
+                    },
+                    "memoryUsedKB": {
+                      "description": "Used memory (calculated as total - free - buffers - cache), in kB.",
+                      "type": "integer",
+                      "format": "int64"
+                    },
+                    "memoryBuffersAndCachedKB": {
+                      "description": "Memory used by kernel buffers (Buffers in /proc/meminfo) and page cache and slabs (Cached and SReclaimable in /proc/meminfo), in kB.",
+                      "type": "integer",
+                      "format": "int64"
+                    },
+                    "memoryFreeKB": {
+                      "description": "Unused memory (MemFree and SwapFree in /proc/meminfo), in kB.",
+                      "type": "integer",
+                      "format": "int64"
+                    },
+                    "memoryTotalKB": {
+                      "description": "Total installed memory (MemTotal and SwapTotal in /proc/meminfo), in kB.",
+                      "type": "integer",
+                      "format": "int64"
+                    },
+                    "cpuUsage": {
+                      "description": "A float representing the current system-wide CPU utilization as a percentage.",
+                      "type": "number",
+                      "format": "double"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "ManagedCassandraReaperStatus": {
+      "type": "object",
+      "properties": {
+        "healthy": {
+          "type": "boolean"
```

Comment:
>boolean [](http://example.com/codeflow?start=19&length=7)

ARM recommends enums over booleans for future proof APIs.

Standard guidance is: replace boolean/switch properties with better enum whenever possible.

A boolean will forever have two valid values (true or false). A string enum is more flexible and future proof option because it allows additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].. Also, properties should always ideally provide more meaningful values just than True and False, which are meaningless without context. For example two boolean switches "isTypeA" and "isTypeB" can be replaced with one enum "type": [A, B, DefaultType].

Note: do NOT just define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its still confusing, thanks to the boolean value names. Try to choose values that are meaningful even without context, like 'Encrypted', 'Unencrypted', 'TlsEncrypted'"